server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -561,29 +557,6 @@ protected void checkNoTimeZone(@Nullable ZoneId timeZone) {
         }
     }
 
-    /**
-     * Extract a {@link Term} from a query created with {@link #termQuery} by
-     * recursively removing {@link BoostQuery} wrappers.
-     * @throws IllegalArgumentException if the wrapped query is not a {@link TermQuery}
-     */
-    public static Term extractTerm(Query termQuery) {
-        while (termQuery instanceof BoostQuery) {
-            termQuery = ((BoostQuery) termQuery).getQuery();
-        }
-        if (termQuery instanceof TermInSetQuery tisQuery) {
-            PrefixCodedTerms terms = tisQuery.getTermData();
-            if (terms.size() == 1) {
-                TermIterator it = terms.iterator();
-                BytesRef term = it.next();
-                return new Term(it.field(), term);
-            }
-        }
-        if (termQuery instanceof TermQuery == false) {
-            throw new IllegalArgumentException("Cannot extract a term from a query of type " + termQuery.getClass() + ": " + termQuery);
```

Comment:
this was previously causing 400 , catching a scenario where match_only_text, which does not store positions, is targeted by a span query. It falls here only because the mapper returns a constant score query which does not get unwrapped. We should rather make sure that there are positions stored for the field that is targeted, or we may get an IllegalStateException from WeightSpan in Lucene which causes a 500 status code.